src/Config/ConfigurationResolver.php

```diff
@@ -0,0 +1,401 @@
+<?php
+
+
+namespace FriendsOfTwig\Twigcs\Config;
+
+use Pimple\Container;
+use Symfony\Component\Filesystem\Filesystem;
+use FriendsOfTwig\Twigcs\Finder;
+use Symfony\Component\Finder\Finder as SymfonyFinder;
+use FriendsOfTwig\Twigcs\Ruleset\RulesetInterface;
+use FriendsOfTwig\Twigcs\Validator\Violation;
+use function fnmatch;
+
+final class ConfigurationResolver
+{
+    const PATH_MODE_OVERRIDE = 'override';
+    const PATH_MODE_INTERSECTION = 'intersection';
+
+    /**
+     * @var null|ConfigInterface
+     */
+    private $config;
+
+    /**
+     * @var null|string
+     */
+    private $configFile;
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    private $path;
+
+    /**
+     * Options which can be set via Cli
+     *
+     * @var array
+     */
+    private $options = [
+        'path' => [],
+        'path-mode' => self::PATH_MODE_OVERRIDE,
+        'severity' => null,
+        'reporter-service-name' => null,
+        'ruleset-class-name' => null,
+        'exclude' => [],
+        'config' => null,
+    ];
+
+    private $finder;
+
+    /**
+     * @var Container
+     */
+    private $container;
+    /**
```

Comment:
Just a tiny thing: maybe add a line above between the comment and the variable?

Reply:
well would be good to define a coding standard that can be automatically applied (with php-cs-fixer or ecs). not knowing what the coding standard is i prefere to have those properties together